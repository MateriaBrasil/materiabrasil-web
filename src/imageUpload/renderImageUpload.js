import React from 'react'

import ImageUpload from './ImageUpload'

export default props => editProps => <ImageUpload {...props} {...editProps} />
