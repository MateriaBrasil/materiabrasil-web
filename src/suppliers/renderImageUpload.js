import render from 'imageUpload/render';

export default props => routeProps => {
  console.log(props);
  const { id } = routeProps.match.params;
  const name = 'suppliers';
  const redirectUrl = `/suppliers/${props.currentUser.suppliers[0].slug}`;

  return render({
    ...props,
    ...routeProps,
    id,
    name,
    redirectUrl,
  });
};
