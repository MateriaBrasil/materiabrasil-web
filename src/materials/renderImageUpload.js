import React from 'react'

import { Edit } from 'croods'

import renderImageUpload from '../imageUpload/renderImageUpload'
import renderUpdated from '../imageUpload/renderUpdated'

export default props => routeProps => {
  const { id } = routeProps.match.params
  const name = 'materials'
  const redirectUrl = `/materials/${id}`

  return (
    <Edit
      id={id}
      name={name}
      render={renderImageUpload({
        ...props,
        ...routeProps,
        id,
        name,
        redirectUrl,
      })}
      renderUpdated={renderUpdated(redirectUrl)}
    />
  )
}
