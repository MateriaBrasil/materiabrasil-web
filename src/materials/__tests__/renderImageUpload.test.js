import React from 'react'
import renderer from 'react-test-renderer'

import renderImageUpload from '../renderImageUpload'

jest.mock('../../imageUpload/renderImageUpload', () => props => (
  <div {...props}>renderImageUpload</div>
))
jest.mock('../../imageUpload/renderUpdated', () => props => (
  <div {...props}>renderUpdated</div>
))
jest.mock('croods', () => ({
  Edit: ({ children, ...props }) => <div {...props}>Edit</div>,
}))

it('renders correctly', () => {
  const routeProps = { match: { params: { id: 1234 } } }
  const Component = renderImageUpload({ foo: 'bar' })

  const tree = renderer.create(<Component {...routeProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})
