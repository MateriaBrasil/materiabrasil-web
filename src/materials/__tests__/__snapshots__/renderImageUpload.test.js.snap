// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id={1234}
  name="materials"
  render={
    <div
      foo="bar"
      id={1234}
      match={
        Object {
          "params": Object {
            "id": 1234,
          },
        }
      }
      name="materials"
      redirectUrl="/1234"
    >
      renderImageUpload
    </div>
  }
  renderUpdated={
    <div
      0="/"
      1="1"
      2="2"
      3="3"
      4="4"
    >
      renderUpdated
    </div>
  }
>
  Edit
</div>
`;
